src/Serialization.js

```diff
@@ -83,10 +87,13 @@ exports.addVkeywitness = witnesses => witness => () => witnesses.add(witness);
 exports.newVkeyFromPublicKey = public_key => () => lib.Vkey.new(public_key);
 
 exports._publicKeyFromBech32 = maybe => bech32 => {
-  try {
-    return maybe.just(lib.PublicKey.from_bech32(bech32));
-  } catch (_) {
-    return maybe.nothing;
+  // this is needed because try/catch overuse breaks runtime badly
+  // https://github.com/Plutonomicon/cardano-transaction-lib/issues/875
+  // prefixes defined in CIP-5: https://github.com/cardano-foundation/CIPs/tree/master/CIP-0005 
+  if (/^(acct|acct_shared|addr|addr_shared|policy|pool|root|root_shared|stake|stake_shared)_x?vk1[0-9a-z]+$/.test(bech32)) {
```

Comment:
You need to find a single prefix. Construct a PublicKey value and call `to_bech32` on it to see the prefix. The simplest way is to add `console.log(key.to_bech32())` and run the test suite.